src/battle/anim/fire.c

```diff
@@ -0,0 +1,112 @@
+#include "global.h"
+#include "battle_anim.h"
+#include "rom_8077ABC.h"
+#include "trig.h"
+
+extern s16 gBattleAnimArgs[];
+extern u8 gAnimBankAttacker;
+
+void sub_80D5038(struct Sprite *);
+void sub_80D50E8(struct Sprite *);
+
+void sub_80D4ED8(struct Sprite *sprite) {
+    sprite->data[0] = gBattleAnimArgs[0];
+    sprite->data[1] = 0x3C;
+    sprite->data[2] = 0x9;
+    sprite->data[3] = 0x1E;
+    sprite->data[4] = 0xFE00;
+    StoreSpriteCallbackInData(sprite, DestroyAnimSprite);
+    sprite->callback = sub_8078174;
+    sub_8078174(sprite);
+}
+
+void sub_80D4F18(struct Sprite *sprite) {
+    sub_807867C(sprite, gBattleAnimArgs[0]);
+    sprite->pos1.y += gBattleAnimArgs[1];
+    sprite->data[0] = gBattleAnimArgs[4];
+    sprite->data[1] = gBattleAnimArgs[2];
+    sprite->data[2] = gBattleAnimArgs[3];
+    sprite->callback = sub_8078394;
+    StoreSpriteCallbackInData(sprite, DestroyAnimSprite);
+}
+
+void sub_80D4F5C(struct Sprite *sprite) {
+    sub_8078650(sprite);
+    if (GetBankSide(gAnimBankAttacker) << 24) {
```

Comment:
The "<< 24" is just the compiler treating GetBankSide's return value as a u8. You want "getBankSide() != 0" here.  That's how it's done most other places, since it's not conceptually a Boolean.  (There are several other places where you're using this shift that can  be updated.